pkg/controller/stackconfigpolicy/elasticsearch_config_secret_mounts_settings.go

```diff
@@ -0,0 +1,186 @@
+				SecretMountsHashAnnotation:        secretsMountHash,
+			},
+		},
+		Data: map[string][]byte{
+			elasticSearchConfigKey: configDataJsonBytes,
+			secretsMountKey:        secretMountBytes,
+		},
+	}
+
+	// Set the Elasticsearch as the soft owner
+	setPolicyAsSoftOwner(&elasticsearchConfigSecret, policy)
+
+	// Add label to delete secret on deletion of the stack config policy
+	elasticsearchConfigSecret.Labels[eslabel.StackConfigPolicyOnDeleteLabelName] = "delete"
+
+	return elasticsearchConfigSecret, nil
+}
+
+func ReconcileElasticsearchConfigSecret(ctx context.Context,
+	c k8s.Client,
+	expected corev1.Secret,
+	es esv1.Elasticsearch,
+	policy policyv1alpha1.StackConfigPolicy) error {
+	reconciled := &corev1.Secret{}
+	return reconciler.ReconcileResource(reconciler.Params{
+		Context:    ctx,
+		Client:     c,
+		Owner:      &es,
+		Expected:   &expected,
+		Reconciled: reconciled,
+		NeedsUpdate: func() bool {
+			return !maps.IsSubset(expected.Labels, reconciled.Labels) ||
+				!maps.IsSubset(expected.Annotations, reconciled.Annotations) ||
+				!reflect.DeepEqual(expected.Data, reconciled.Data)
+		},
+		UpdateReconciled: func() {
+			reconciled.Labels = maps.Merge(reconciled.Labels, expected.Labels)
+			// remove managed labels if they are no longer defined
+			for _, label := range managedLabels {
+				if _, ok := expected.Labels[label]; !ok {
+					delete(reconciled.Labels, label)
+				}
+			}
+			reconciled.Annotations = maps.Merge(reconciled.Annotations, expected.Annotations)
+			reconciled.Data = expected.Data
+		},
+	})
+}
+
+// setSoftOwner sets the given stack config policy as soft owner of the Secret using the "softOwned" labels.
+func setPolicyAsSoftOwner(secret *corev1.Secret, policy policyv1alpha1.StackConfigPolicy) {
+	if secret.Labels == nil {
+		secret.Labels = map[string]string{}
+	}
+	secret.Labels[reconciler.SoftOwnerNamespaceLabel] = policy.GetNamespace()
+	secret.Labels[reconciler.SoftOwnerNameLabel] = policy.GetName()
+	secret.Labels[reconciler.SoftOwnerKindLabel] = policy.GetObjectKind().GroupVersionKind().Kind
+}
+
+// reconcileSecretMountSecretsESNamespace creates the secrets in SecretMounts to the respective Elasticsearch namespace where they should be mounted to.
+func reconcileSecretMountSecretsESNamespace(ctx context.Context, c k8s.Client, es esv1.Elasticsearch, policy *policyv1alpha1.StackConfigPolicy) error {
+	for i, secretMount := range policy.Spec.Elasticsearch.SecretMounts {
+		additionalSecret := corev1.Secret{}
+		namespacedName := types.NamespacedName{
+			Name: secretMount.SecretName,
+			// TODO: should this be policy namespace or operator's namespace
+			Namespace: policy.Namespace,
+		}
+		if err := c.Get(ctx, namespacedName, &additionalSecret); err != nil {
+			return err
+		}
+
+		// Recreate it in the Elasticsearch namespace, prefix with es name.
+		secretName := esv1.ESNamer.Suffix(es.Name, additionalSecret.Name)
+		// Replace the name in the policy object as well
```

Comment:
Why do we need that side effect? (which feels a bit wrong/hacky tbh)

Reply:
I was doing this so that I can read the secret name from the policy while creating the volumes for the pod, but I made the change to just generate the secret name to look at, removed this.